tests/envs/functional/test_core.py

```diff
@@ -12,23 +12,25 @@ def __init__(self, options: Optional[Dict[str, Any]] = None):
     def initial(self, rng: Any) -> np.ndarray:
         return np.array([0, 0], dtype=np.float32)
 
-    def observation(self, state: np.ndarray) -> np.ndarray:
+    def observation(self, state: np.ndarray, rng: Any) -> np.ndarray:
         return state
 
     def transition(self, state: np.ndarray, action: int, rng: None) -> np.ndarray:
         return state + np.array([0, action], dtype=np.float32)
 
-    def reward(self, state: np.ndarray, action: int, next_state: np.ndarray) -> float:
+    def reward(
+        self, state: np.ndarray, action: int, next_state: np.ndarray, rng: Any
+    ) -> float:
         return 1.0 if next_state[1] > 0 else 0.0
 
-    def terminal(self, state: np.ndarray) -> bool:
+    def terminal(self, state: np.ndarray, rng: Any) -> bool:
         return state[1] > 0
 
 
 def test_api():
     env = BasicTestEnv()
     state = env.initial(None)
-    obs = env.observation(state)
+    obs = env.observation(state, None)
```

Comment:
Can we make the rng argument optional the same way as params?

Reply:
How should `rng` default be handled?
If an environment's needs `transition` member function requires `rng`, what should be done?

In the case of parameters, there is clearly a default

Reply:
@RedTachyon